src/main/java/io/github/ryotan/code/constraint/CodeValue.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code.constraint;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Predicate;
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+
+import io.github.ryotan.code.Code;
+import io.github.ryotan.code.CodeEnum;
+import io.github.ryotan.code.CodeEnum.Filters;
+import io.github.ryotan.code.util.CodeEnumReflectionUtil;
+
+/**
+ * {@link CodeEnum}の値であることをあらわす制約です。
+ * <p>
+ * {@link String}あるいは{@link CodeEnum}の型のフィールドに対して{@link CodeValue}アノテーションを付けておくと、
+ * {@link CodeValue#value()}に指定した{@code CodeEnum}に含まれるかどうかの精査が実行されます。
+ * 制約を満たさない場合は、{@link CodeValue#message()}に指定されたメッセージの{@link javax.validation.ConstraintViolation}を生成します。
+ * デフォルトのメッセージは、{@value DEFAULT_MESSAGE}です。
+ * </p>
+ * <p>
+ * {@link CodeValue#filters()}を指定した場合、指定されたフィルタでフィルタリングされたコード値に精査対象の値が含まれるかどうかを精査します。
+ * </p>
+ * <p>
+ * {@link CodeEnum}型に対してこの制約を利用する場合、{@link CodeValue#value()}は、対象の型と同じクラスを指定してください。
+ * </p>
+ *
+ * @author ryotan
+ */
+@Target({METHOD, FIELD, PARAMETER})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = {CodeValue.CodeValueValidator.class, CodeValue.CodeEnumValidator.class})
+public @interface CodeValue {
+    /**
+     * デフォルトのエラーメッセージ。
+     */
+    String DEFAULT_MESSAGE = "{io.github.ryotan.code.constraint.CodeValue.message}";
+
+    /**
+     * 指定されたコード値が{@code value}で指定されたコードのコード値に含まれない場合のエラーメッセージを指定してください。
+     *
+     * @return エラーメッセージ
+     * @see Constraint
+     */
+    String message() default DEFAULT_MESSAGE;
+
+    /**
+     * @return 精査対象となるグループの配列
+     * @see Constraint
+     */
+    Class<?>[] groups() default {};
+
+    /**
+     * @return {@link Payload}s for extensibility purposes
+     * @see Constraint
+     */
+    Class<? extends Payload>[] payload() default {};
+
+    /**
+     * @return 制約の対象とする {@link CodeEnum}。
+     */
+    Class<? extends CodeEnum<?>> value();
+
+    /**
+     * 制約の対象とする{@link CodeEnum}をフィルタリングするためのメソッド名を指定します。複数指定された場合、AND条件として扱われます。
+     * <p>
+     * メソッド名には、{@link io.github.ryotan.code.CodeEnum.Filter}アノテーションが付与されているメソッドを指定してください。
+     * </p>
+     *
+     * @return フィルタメソッド名の配列
+     */
+    String[] filters() default {};
+
+    /**
+     * {@link CodeValue}の制約を満しているかどうかを精査する{@link ConstraintValidator}用のサポートクラスです。
+     *
+     * @param <T> 精査対象の型
+     *
+     * @author ryotan
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [At-clauses have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'.](https://www.codacy.com/app/ryotan/code-enum-support/file/3991998002/issues/source?bid=3375378&fileBranchId=3375797#l96)